src/native/eventpipe/ep-config.c

```diff
@@ -391,8 +391,8 @@ ep_config_build_event_metadata_event (
 	const ep_char16_t *provider_name_utf16 = ep_provider_get_provider_name_utf16 (provider);
 	const uint8_t *payload_data = ep_event_get_metadata (source_event);
 	uint32_t payload_data_len = ep_event_get_metadata_len (source_event);
-	uint32_t provider_name_len = (uint32_t)((ep_rt_utf16_string_len (provider_name_utf16) + 1) * sizeof (ep_char16_t));
-	uint32_t instance_payload_size = sizeof (metadata_id) + provider_name_len + payload_data_len;
+	uint32_t provider_name_len = (uint32_t)ep_rt_utf16_string_len (provider_name_utf16);
```

Comment:
Are we missing NULL terminating character here? We had that in the previous used length.

Reply:
Right, the length does not contain the null termination character anymore since I changed `ep_write_buffer_string_utf16_t` where the length parameter should describe the string length. I adapted `instance_payload_size` instead, now.